common/src/primitives/height.rs

```diff
@@ -5,7 +5,19 @@ use std::ops::{Add, Sub};
 type HeightIntType = u64;
 type DistanceIntType = i64;
 
-#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Encode, Decode)]
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    PartialOrd,
+    Ord,
+    PartialEq,
+    Eq,
+    Encode,
+    Decode,
+    serde::Serialize,
```

Comment:
This is the first file I see in this PR, and my internal reaction was: "Nooooooooooooooooooooooooooooooooooooo!"

https://www.youtube.com/watch?v=yEMimY0HyUk

Reply:
On a serious note, this should have a custom implementation, probably to avoid having the tuple affect the json representation. I'm not sure whether serde will map the tuple to json or will do something reasonable to flatten the data in it.

Reply:
A unit tuple struct is represented as the contained element by `serde` in json. So in json, `BlockHeight` will be just a number. That seems sensible to me, do we want something else?

Reply:
Perfect. I thought it would try something like `{ "0": 1234 }`.